src/main/java/com/fasterxml/jackson/databind/util/ExceptionUtil.java

```diff
@@ -49,6 +49,12 @@ private static boolean isFatal(Throwable throwable) {
         return (throwable instanceof VirtualMachineError
                 || throwable instanceof ThreadDeath
                 || throwable instanceof InterruptedException
-                || throwable instanceof LinkageError);
+                || throwable instanceof ClassCircularityError
+                || throwable instanceof ClassFormatError
+                || throwable instanceof ExceptionInInitializerError
```

Comment:
Also not sure this belongs here -- isn't this a wrapper used around other failures, during class initialization?
And as such likely failure mode when trying to load something that depends on something else not available.

Others make sense.

Reply:
which Error is not fatal? all the new ones or one in particular?